### PR TITLE
Fix Binary Signing

### LINKED ISSUE #5116: Fix Binary Signing

### PR DESCRIPTION
**Pull request by ugras-ergun-sonarsource**
Fixes #5116

### CODE REVIEW

File: build/Signing.targets [hunk 1]
```diff
@@ -40,25 +40,13 @@
     <CallTarget Targets="ValidateCommonSigningInputs" />
     <Error Condition="@(VSIXesToSign) == '' " Text="The list of VSIXes to sign is empty." />
     
-    <!-- For info on timestamping see https://www.digicert.com/kb/code-signing/signcode-signtool-command-line.htm -->
```
Comment (ugras-ergun-sonarsource) [inline build/Signing.targets:43]: Link is dead

File: build/Signing.targets [hunk 2]
```diff
@@ -40,25 +40,13 @@
...
     <Message Importance="high" Text="Assembly to sign: %(AssembliesToSign.FullPath)" />
-    <CallTarget Targets="ValidateCommonSigningInputs" />
     <Error Condition=" @(AssembliesToSign) == '' " Text="The list of assemblies to sign is empty." />
-    <Error Condition=" $(SIGNTOOL_PATH) == '' " Text="The location of the signtool.exe has not been set ('SIGNTOOL_PATH')" />
-    <Error Condition=" !Exists($(SIGNTOOL_PATH)) " Text="Signing tool exe does not exist at the specified location: $(SIGNTOOL_PATH)" />
-
-    <Exec Command="&quot;$(SIGNTOOL_PATH)&quot; sign /fd SHA256 /sha1 $(pfxSha1) /f &quot;$(pfxCertificatePath)&quot; /p $(pfxPassword) /tr http://timestamp.digicert.com?alg=sha256 /td SHA256 &quot;%(AssembliesToSign.FullPath)&quot;" />
-  </Target>
 
-  <Target Name="ValidateCommonSigningInputs">
```
Comment (ugras-ergun-sonarsource) [inline build/Signing.targets:57]: no need for this anymore we don't have these.

File: build/Signing.targets [hunk 3]
```diff
@@ -40,25 +40,13 @@
...
     
-    <!-- For info on timestamping see https://www.digicert.com/kb/code-signing/signcode-signtool-command-line.htm -->
     <Exec Command="$(VsixSignToolPath) sign /f &quot;$(pfxCertificatePath)&quot; /p $(pfxPassword) /sha1 $(pfxSha1) /tr http://timestamp.digicert.com?alg=sha256 /td SHA256 /v &quot;%(VSIXesToSign.Identity)&quot;" />
   </Target>
 
   <Target Name="SignAssemblies">
     <Message Importance="high" Text="Assembly to sign: %(AssembliesToSign.FullPath)" />
-    <CallTarget Targets="ValidateCommonSigningInputs" />
     <Error Condition=" @(AssembliesToSign) == '' " Text="The list of assemblies to sign is empty." />
-    <Error Condition=" $(SIGNTOOL_PATH) == '' " Text="The location of the signtool.exe has not been set ('SIGNTOOL_PATH')" />
```
Comment (ugras-ergun-sonarsource) [inline build/Signing.targets:51]: we do not calculate signtool path anymore

File: src/Integration.Vsix/Integration.Vsix.csproj [hunk 4]
```diff
@@ -365,8 +365,8 @@
     <Message Importance="high" Text="Copying vsix from $(TargetDir)$(TargetVsixContainerName) to $(BinariesFolder)$(TargetVsixContainerName)" />
     <Copy OverwriteReadOnlyFiles="true" SourceFiles="$(TargetDir)$(TargetVsixContainerName)" DestinationFiles="$(BinariesFolder)$(TargetVsixContainerName)" />
 
-    <Message Condition=" $(SignArtifacts) != 'true' " Importance="high" Text="Skipping signing VSIXes - SignArtifacts = '$(SignArtifacts)'" />
-    <CallTarget Condition=" $(SignArtifacts) == 'true' " Targets="LocateVsixSignToolAndBinaries;SignVsixes" />
+    <!--<Message Condition=" $(SignArtifacts) != 'true' " Importance="high" Text="Skipping signing VSIXes - SignArtifacts = '$(SignArtifacts)'" />
```
Comment (ugras-ergun-sonarsource) [inline src/Integration.Vsix/Integration.Vsix.csproj:368]: signing vsix does not work so we have removed this step but we will add it later.

File: src/TestInfrastructure/TestInfrastructure.csproj [hunk 5]
```diff
@@ -8,6 +8,7 @@
          We'd prefer to set this the shared SonarLint.VSSpecificAssemblies.props file, but
          the IDE isn't picking up the setting if we do that. -->
     <DefineConstants Condition="$(VSTargetVersion)=='2022'">$(DefineConstants);VS2022</DefineConstants>
+    <RequiresAlmAuthentication>true</RequiresAlmAuthentication>
```
Comment (ugras-ergun-sonarsource) [inline src/TestInfrastructure/TestInfrastructure.csproj:11]: this is the step that changed the testing failures.

**Review by georgii-borovinskikh-sonarsource — APPROVED** [review]

**Pull request merged**